src/Magnum/Platform/GlfwApplication.h

```diff
@@ -0,0 +1,652 @@
+    protected:
+    #else
+    private:
+    #endif
+        /** @copydoc Sdl2Application::viewportEvent() */
+        virtual void viewportEvent(const Vector2i& size);
+
+        /** @copydoc Sdl2Application::drawEvent() */
+        virtual void drawEvent() = 0;
+
+        /*@}*/
+
+        /** @{ @name Keyboard handling */
+
+        /** @copydoc Sdl2Application::keyPressEvent() */
+        virtual void keyPressEvent(KeyEvent& event);
+
+        /** @copydoc Sdl2Application::keyReleaseEvent() */
+        virtual void keyReleaseEvent(KeyEvent& event);
+
+        /*@}*/
+
+        /** @{ @name Mouse handling */
+
+    public:
+        /**
+         * @brief Mouse cursor
+         *
+         * @see @ref setMouseCursor()
+         */
+        enum class MouseCursor: int {
+            Default = GLFW_CURSOR_NORMAL,  /**< Default cursor provided by parent window */
+            Hidden = GLFW_CURSOR_HIDDEN,   /**< Hidden cursor */
+            None = GLFW_CURSOR_DISABLED    /**< No cursor */
+        };
+
+        /** @brief Warp mouse cursor to given coordinates */
+        void warpCursor(const Vector2i& position) {
+            glfwSetCursorPos(_window, Double(position.x()), Double(position.y()));
+        }
+
+    #ifdef DOXYGEN_GENERATING_OUTPUT
+    protected:
+    #else
+    private:
+    #endif
+        /** @copydoc Sdl2Application::mousePressEvent() */
+        virtual void mousePressEvent(MouseEvent& event);
+
+        /** @copydoc Sdl2Application::mouseReleaseEvent() */
+        virtual void mouseReleaseEvent(MouseEvent& event);
+
+        /**
+         * @brief Mouse move event
+         *
+         * Called when any mouse button is pressed and mouse is moved. Default
+         * implementation does nothing.
+         * @see @ref setMouseTracking()
+         */
+        virtual void mouseMoveEvent(MouseMoveEvent& event);
+
+        /*@}*/
+
+    private:
+        static void staticViewportEvent(GLFWwindow*, int w, int h) {
+            _instance->viewportEvent({w, h});
+        }
+
+        static void staticKeyEvent(GLFWwindow* window, int key, int scancode, int action, int mod);
+
+        static void staticMouseEvent(GLFWwindow* window, int button, int action, int mods);
+
+        static void staticMouseMoveEvent(GLFWwindow* window, double x, double y);
+
+        static void staticErrorCallback(int error, const char* description);
+
+        static GlfwApplication* _instance;
+		
+		GLFWwindow* _window;
+
+        std::unique_ptr<Platform::Context> _context;
+
+        bool _needsRedraw;
+};
+
+/**
+@brief Configuration
+
+Double-buffered RGBA window with depth and stencil buffers.
+@see @ref GlfwApplication(), @ref createContext(), @ref tryCreateContext()
+*/
+class GlfwApplication::Configuration {
+    public:
+        /**
+         * @brief Context flag
+         *
+         * @see @ref Flags, @ref setFlags()
+         */
+        enum class Flag: int {
+            NoError = GLFW_CONTEXT_NO_ERROR,  /**< Create debug context */
```

Comment:
Huh, GLFW supports `KHR_no_error` already? Wow. I don't even have the extension in the list yet :)

(The comment is misleading, though.)